packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -0,0 +1,201 @@
+		return select( 'core/block-editor' ).getSettings().mediaUpload;
+	} );
+	const editMediaButtonRef = createRef();
+
+	const stopPropagation = ( event ) => {
+		event.stopPropagation();
+	};
+
+	const stopPropagationRelevantKeys = ( event ) => {
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+			event.stopPropagation();
+		}
+	};
+
+	const selectMedia = ( media ) => {
+		onSelect( media );
+		setMediaURLValue( media.url );
+		speak( __( 'The media file has been replaced' ) );
+	};
+
+	const uploadFiles = ( event ) => {
+		const files = event.target.files;
+		const setMedia = ( [ media ] ) => {
+			setShowMediaReplaceOptions( false );
+			selectMedia( media );
+		};
+		mediaUpload( {
+			allowedTypes,
+			filesList: files,
+			onFileChange: setMedia,
+			onError,
+		} );
+	};
+
+	const onClose = () => {
+		editMediaButtonRef.current.focus();
+	};
+
+	const onClickOutside = () => ( setShowMediaReplaceOptions( false ) );
+
+	const openOnArrowDown = ( event ) => {
+		if ( event.keyCode === DOWN ) {
+			event.preventDefault();
+			event.stopPropagation();
+			event.target.click();
+		}
+	};
+
+	let urlInputUIContent;
+	if ( showEditURLInput ) {
+		urlInputUIContent = (
+			<LinkEditor
+				onKeyDown={ stopPropagationRelevantKeys }
+				onKeyPress={ stopPropagation }
+				value={ mediaURLValue }
+				isFullWidthInput={ true }
+				hasInputBorder={ true }
+				onChangeInputValue={ ( url ) => ( setMediaURLValue( url ) ) }
+				onSubmit={ ( event ) => {
+					event.preventDefault();
+					onSelectURL( mediaURLValue );
+					setShowEditURLInput( ! showEditURLInput );
+					editMediaButtonRef.current.focus();
+				} }
+			/>
+		);
+	} else {
+		urlInputUIContent = (
+			<LinkViewer
+				isFullWidth={ true }
+				className="editor-format-toolbar__link-container-content block-editor-format-toolbar__link-container-content"
+				url={ mediaURLValue }
+				onEditLinkClick={ () => ( setShowEditURLInput( ! showEditURLInput ) ) }
+			/>
+		);
+	}
+
+	return (
+		<MediaUpload
+			onSelect={ ( media ) => selectMedia( media ) }
+			onClose={ () => setShowMediaReplaceOptions( true ) }
+			allowedTypes={ allowedTypes }
+			render={ ( { open } ) => (
+				<Toolbar className={ 'media-replace-flow components-dropdown-menu' }>
+					<Button
+						ref={ editMediaButtonRef }
+						className={ 'components-icon-button components-dropdown-menu__toggle' }
+						onClick={ () => {
+							setShowMediaReplaceOptions( ! showMediaReplaceOptions );
+						} }
+						onKeyDown={ openOnArrowDown }
+					>
+						<span className="components-dropdown-menu__label" > { name } </span>
+						<span className="components-dropdown-menu__indicator" />
+					</Button>
+					{ showMediaReplaceOptions &&
+						<Popover
+							onClickOutside={ onClickOutside }
+							onClose={ onClose }
+							className={ 'media-replace-flow__options' }
+						>
+							<>
```

Comment:
Fragments are only needed if there are multiple siblings being rendered at the root level of a hierarchy

As this isn't the root level (`<Popover>` is wrapping) the fragment can be removed. I think it would be beneficial as it'll reduce the amount of nesting.